### PR TITLE
YANG model for FABRIC_MONITOR|FABRIC_MONITOR_DATA

### PR DESCRIPTION
**Pull request by jfeng-arista**
#### Why I did it
Table FABRIC_MONITOR|FABRIC_MONITOR_DATA in CONFIG_DB was added in https://github.com/sonic-net/sonic-buildimage/pull/14170 
The YANG model for that table is still missing. 

Need to wait https://github.com/sonic-net/sonic-buildimage/pull/14170 get merged first.

#### How I did it
Added new YANG model FABRIC_MONITOR and associated unit tests.

#### How to verify it
Passing unit tests

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    container sonic-fabric-monitor {
+
+        container FABRIC_MONITOR {
+
+            description "FABRIC_MONITOR part of config_db.json";
+
+            container FABRIC_MONITOR_DATA {
+
+                leaf monErrThreshCrcCells {
```
Comment (arlakshm) [inline src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:22]: can we a description to the leaf nodes

Comment (jfeng-arista) [reply src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:22]: Added

**Review by arlakshm — APPROVED** [review]

**Comment by gechiang** [comment]
@jfeng-arista , @arlakshm , Should this PR be labeled for chassis branch as well as 202305?  Any other branch should this be picked up as well?

**Comment by gechiang** [comment]
@jfeng-arista , I am not able to cleanly cherry-pick this into the "https://github.com/Azure/sonic-buildimage-msft.git" 202205.
Can you please create a new PR in "https://github.com/Azure/sonic-buildimage-msft.git" repo under the 202205 branch for this exact change and tag me to review/approve/merge.
Thanks!

**Comment by jfeng-arista** [comment]
will do, thank you

**Pull request merged**